pubnub/event_engine/manage_effects.py

```diff
@@ -63,7 +65,12 @@ def run(self):
     async def handshake_async(self, channels, groups, stop_event):
         request = Subscribe(self.pubnub).channels(channels).channel_groups(groups).cancellation_event(stop_event)
         handshake = await request.future()
-        if not handshake.status.error:
+
+        if handshake.status.error:
+            logging.warning(f'Handshake failed: {handshake.status.error_data.__dict__}')
+            handshake_failure = events.HandshakeFailureEvent(handshake.status.error_data, 1)
+            self.event_engine.trigger(handshake_failure)
+        else:
             cursor = handshake.result['t']
             timetoken = cursor['t']
             region = cursor['r']
```

Comment:
I see that much code in python SDK has many nested places. 
Maybe it's not the best place to start doing it but can you consider using this syntax?
```suggestion

        if handshake.status.error:
            logging.warning(f'Handshake failed: {handshake.status.error_data.__dict__}')
            handshake_failure = events.HandshakeFailureEvent(handshake.status.error_data, 1)
            self.event_engine.trigger(handshake_failure)
            return
       cursor = handshake.result['t']
       timetoken = cursor['t']
       region = cursor['r']
```